src/SoftDeleteable/Query/TreeWalker/SoftDeleteableWalker.php

```diff
@@ -118,10 +161,7 @@ private function extractComponents(array $queryComponents)
     {
         $em = $this->getEntityManager();
 
-        foreach ($queryComponents as $alias => $comp) {
-            if (!isset($comp['metadata'])) {
```

Comment:
From a practical perspective, the `metadata` key cannot be missing as the parent class' `setQueryComponent()` method requires it to be present.

Reply:
It would be interesting maybe removing the parameter of the method and call `$this->getQueryComponents()` here (to [get the array shape information](https://github.com/doctrine/orm/blob/ca0a6bbf71c5b475291d47cca92925f149f48138/lib/Doctrine/ORM/Query/TreeWalker.php#L28-L36)) or maybe add the array shape as parameter.

Reply:
I agree. That's why it's important to include SCA in our CI pipeline.